shell/platform/common/cpp/client_wrapper/include/flutter/encodable_value.h

```diff
@@ -6,16 +6,150 @@
+// should update your code as soon as possible to use the new std::variant
+// version, or it will break when the legacy version is removed.
+#ifndef USE_LEGACY_ENCODABLE_VALUE
+
+class EncodableValue;
+
+// Convenience type aliases.
+using EncodableList = std::vector<EncodableValue>;
+using EncodableMap = std::map<EncodableValue, EncodableValue>;
+
+// An object that can contain any value or collection type supported by
+// Flutter's standard method codec.
+//
+// For details, see:
+// https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html
+//
+// As an example, the following Dart structure:
+//   {
+//     'flag': true,
+//     'name': 'Thing',
+//     'values': [1, 2.0, 4],
+//   }
+// would correspond to:
+//   EncodableValue(EncodableMap{
+//       {EncodableValue("flag"), EncodableValue(true)},
+//       {EncodableValue("name"), EncodableValue("Thing")},
+//       {EncodableValue("values"), EncodableValue(EncodableList{
+//                                      EncodableValue(1),
+//                                      EncodableValue(2.0),
+//                                      EncodableValue(4),
+//                                  })},
+//   })
+//
+// The primary API surface for this object is std::variant. For instance,
+// getting a string value from an EncodableValue, with type checking:
+//   if (std::get<std::string>(value)) {
+//     std::string some_string = std::get<std::string>(value);
+//   }
+//
+// The order/indexes of the variant types is part of the API surface, and is
+// guaranteed not to change.
+class EncodableValue : public std::variant<std::monostate,
+                                           bool,
+                                           int32_t,
+                                           int64_t,
+                                           double,
+                                           std::string,
+                                           std::vector<uint8_t>,
+                                           std::vector<int32_t>,
+                                           std::vector<int64_t>,
+                                           std::vector<double>,
+                                           EncodableList,
+                                           EncodableMap> {
+ public:
+  // Rely on std::variant for most of the constructors/operators.
+  using super = std::variant<std::monostate,
+                             bool,
+                             int32_t,
+                             int64_t,
+                             double,
+                             std::string,
+                             std::vector<uint8_t>,
+                             std::vector<int32_t>,
+                             std::vector<int64_t>,
+                             std::vector<double>,
+                             EncodableList,
+                             EncodableMap>;
+  using super::super;
+  using super::operator=;
+
+  explicit EncodableValue() : super() {}
```

Comment:
This is not necessary because the monostate is already part of the variant. `= default` should be fine.

Reply:
Good point, will change to `= default`. (I was surprised I needed to declare it at all, vs. having it come in through the `using`, actually).